terraform/jupyterhub.tf

```diff
@@ -1,74 +1,12 @@
-resource "aws_iam_role" "ecs-service" {
-  assume_role_policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Action": "sts:AssumeRole",
-            "Principal": {
-                "Service": "ecs.amazonaws.com"
-            },
-            "Effect": "Allow",
-            "Sid": ""
-        }
-    ]
-}
-EOF
-}
-
-resource "aws_iam_role_policy_attachment" "ecs-service" {
-  role       = "${aws_iam_role.ecs-service.name}"
-  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerServiceFullAccess"
-}
-
-resource "aws_iam_instance_profile" "ecs-service" {
-  role = "${aws_iam_role.ecs-service.name}"
-}
-
-resource "aws_launch_configuration" "jupyterhub" {
+resource "aws_spot_instance_request" "cheap_worker" {
+  ami                  = "${var.ecs_ami}"
   iam_instance_profile = "${var.ecs_instance_profile}"
-  image_id             = "${var.ecs_ami}"
   instance_type        = "m3.xlarge"
   key_name             = "${var.key_name}"
-  security_groups      = ["${aws_security_group.security-group.id}"]
+  security_groups      = ["${aws_security_group.security-group.name}"]
```

Comment:
I believe that this is supposed to be IDs.

Reply:
It is actually kind of strange: using the `id` here causes a runtime error, you have to use `name`.  I have noticed a number of these types of inconsistencies.